### PR TITLE
use SO_PEERCRED instead of manual cred writing

### PR DESCRIPTION
**Pull request by aktau**
This saves some finnicky code, and resolves a TODO. The tests I'm adding in another test failed if I just removed the `writeCredentials` step, but now with `SO_PEERCRED` they pass.

`SO_PEERCRED` is supported on BSD and Linux, but effectively only support Linux anyway.

### CODE REVIEW

File: cmd/perflock/creds.go [hunk 1]
```diff
@@ -5,61 +5,15 @@
...
-	scms, err := syscall.ParseSocketControlMessage(oob[:oobn])
-	if err != nil {
-		return nil, err
-	}
-	if len(scms) != 1 {
-		return nil, fmt.Errorf("expected 1 control message, got %d", len(scms))
-	}
-	return syscall.ParseUnixCredentials(&scms[0])
+	defer f.Close()
+	return syscall.GetsockoptUcred(int(f.Fd()), syscall.SOL_SOCKET, syscall.SO_PEERCRED)
```
Comment (aclements) [inline cmd/perflock/creds.go]: I believe people use perflock on macOS (just without the governor stuff). Does SO_PEERCRED work there?

Comment (aktau) [reply cmd/perflock/creds.go]: Huh, it seems less useful to me without being able to affect the CPU governor, disabling it was decisive in  correcting a misleading benchmark lately.

At any rate, I looked around and I think `SO_PEERCRED` may not work, but macOS appears to have something very similar called `LOCAL_PEERCRED`:

- Example: https://lists.gnu.org/archive/html/guix-commits/2015-01/msg01008.html
- macOS offers a libc function called [getpeereid(3)](https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/getpeereid.3.html) apparently built on top of it.

Comment (aktau) [reply cmd/perflock/creds.go]: I found https://github.com/golang/go/issues/27613, which lead me to https://pkg.go.dev/inet.af/peercred, which is basically exactly what we need. I don't know how you feel about dependencies, but this is a tiny one, and it does basically exactly what I'd do, but more completely. I've added another commit using it. PTAL.

**Comment by aktau** [comment]
I don't really know how Go modules work. Should I add the `go.sum` file to the commit?

```
$ cat go.sum
golang.org/x/sys v0.0.0-20210301091718-77cc2087c03b h1:kHlr0tATeLRMEiZJu5CknOw/E8V6h69sXXQFGoPtjcc=
golang.org/x/sys v0.0.0-20210301091718-77cc2087c03b/go.mod h1:h1NjWce9XRLGQEsW7wpKNCjG9DtNlClVuFLEZdDNbEs=
golang.org/x/sys v0.16.0 h1:xWw16ngr6ZMtmxDyKyIgsE93KNKz5HKmMa3b8ALHidU=
golang.org/x/sys v0.16.0/go.mod h1:/VUhepiaJMQUp4+oa/7Zr1D23ma6VTLIYjOOTFZPUcA=
inet.af/peercred v0.0.0-20210906144145-0893ea02156a h1:qdkS8Q5/i10xU2ArJMKYhVa1DORzBfYS/qA2UK2jheg=
inet.af/peercred v0.0.0-20210906144145-0893ea02156a/go.mod h1:FjawnflS/udxX+SvpsMgZfdqx2aykOlkISeAsADi5IU=

```

**Comment by aclements** [comment]
Yep, you should add the `go.sum` file. Also try running `go mod tidy`. I'm not positive it will change anything, but it might. Otherwise LGTM.

**Comment by aktau** [comment]
OK, I've added `go.sum` after running `go mod tidy`. I've also squashed both commits together now that I know you're ok with using package `peercred`.

**Comment by aktau** [comment]
I've rebased on top of pull #9 which has been merged. I've run the tests and they succeed. Ready to merge if you like it.

**Pull request merged**